build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+N_CORES=$(sysctl -n hw.ncpu) # macOS
+# TODO: get cores linux
+
+# ESP-IDF v3.x
+esp_idf_v3_build_native() {
+    cp arduinojason.mk components/ArduinoJson/component.mk
+    make all -j$N_CORES;
+    rm components/ArduinoJson/component.mk
+}
```

Comment:
I think it's probably best to scrap native at this point. Any inconsistency in the installed ESP-IDF version and we're in trouble, as it tries to reconfigure:

```
MacBook-Pro ~/Sites/soulmate/core(feature/redo-project-layout) • ./build.sh
GENCONFIG
*
* Restart config...
*
*
* TCP
*
Enable TCP ISN Hook (LWIP_TCP_ISN_HOOK) [Y/n/?] (NEW) ^C
MacBook-Pro ~/Sites/soulmate/core(feature/redo-project-layout) • ./build.sh docker
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Adding ESP-IDF tools to PATH...
Checking if Python packages are up to date...
Python requirements from /opt/esp/idf/requirements.txt are satisfied.
Added the following directories to PATH:
```

Reply:
It won't share cached files between native and docker builds. I want to see if I can fix this. It's nice to have the option to build natively.

Reply:
We could save which type was built last, and blow away the cache next time - or even change the output directory to `build-docker`?

Reply:
Oh - can we exclude the build directory from the volume mount? That would make compilation a lot faster too. Maybe we have to set the BUILD_OUTPUT_DIR to something outside the volume for that to work?

I think that works by specifying another mount directory but excluding a host folder:

`-v /{path}/build` in build.sh should do it. Note no colon prefix with the local dir